src/main/java/org/folio/rest/impl/RequestObjectFactory.java

```diff
@@ -51,32 +61,45 @@
       });
   }
 
-  private CompletableFuture<RequestType> getRequestType(String patronId, String itemId) {
-    final var itemRepository = new ItemRepository(httpClient);
-    final var userRepository = new UserRepository(httpClient);
+  private CompletableFuture<RequestContext> fetchItem(RequestContext requestContext) {
+    return itemRepository.getItem(requestContext.getItemId(), okapiHeaders)
+      .thenApply(requestContext::setItem);
+  }
 
-    CompletableFuture<JsonObject> userFuture = userRepository.getUser(patronId, okapiHeaders);
-    CompletableFuture<JsonObject> itemFuture = itemRepository.getItem(itemId, okapiHeaders);
+  private CompletableFuture<RequestContext> fetchUser(RequestContext requestContext) {
+    return userRepository.getUser(requestContext.getPatronId(), okapiHeaders)
+      .thenApply(requestContext::setUser);
+  }
 
+  private CompletableFuture<RequestContext> getRequestType(RequestContext requestContext) {
     RequestTypeParameters requestTypeParams = new RequestTypeParameters();
 
-    return CompletableFuture.allOf(userFuture, itemFuture)
-      .thenApply(x -> createRequestPolicyIdCriteria(itemFuture, userFuture, requestTypeParams))
+    return CompletableFuture.completedFuture(requestContext)
```

Comment:
There is no point in wrapping `requestContext` into a Future unless you use it in the next step

Reply:
thanks, fixed